feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-base.component.ts

```diff
@@ -358,4 +396,58 @@ export class ConfiguratorAttributeBaseComponent {
       !this.isReadOnly(attribute)
     );
   }
+
+  /**
+   * Merges value price data received via @see {ConfiguratorValuePriceChanged} events into the given value, if available.
+   * @param {Configurator.Value} value the value
+   * @returns {Configurator.Value} the new value with pricing
+   */
+  protected mergePriceAndValue(value: Configurator.Value): Configurator.Value {
+    const valueName = value.name;
+    if (valueName && this.valuePrices[valueName]) {
+      value = { ...value, valuePrice: this.valuePrices[valueName] };
+    }
+    return value;
+  }
+
+  /**
+   * Creates a text describing the current attribute that can be used as ARIA label.
+   * Includes price information. If a total price is available this price will be used,
+   * otherwise it falls back to the value price, or if no price is available,
+   * no price information will be included in the text.
+   *
+   * @param {Configurator.Attribute} attribute the attribute
+   * @param {Configurator.Value} value the value
+   * @returns {string} translated text
+   */
+  protected getAriaLabelGeneric(
```

Comment:
Could you add missing tests= 
![image](https://github.com/SAP/spartacus/assets/61147963/b86cc6b1-82af-4c1a-8f09-d52d52a00c02)